src/hotspot/share/memory/metaspace.cpp

```diff
@@ -703,13 +703,13 @@ void Metaspace::global_initialize() {
     // case (b)
     ReservedSpace rs;
 
-    // If UseCompressedOops=1, java heap may have been placed in coops-friendly
-    //  territory already (lower address regions), so we attempt to place ccs
+    // If UseCompressedOops=1 and the java heap has been placed in coops-friendly
```

Comment:
s/has/had

Reply:
I'm using [present perfect](https://en.wikipedia.org/wiki/Present_perfect). You suggest to use past perfect (aka [pluperfect](https://en.wikipedia.org/wiki/Pluperfect)). Is that correct so far? If so, then I'd think present perfect is the better fit, isn't it?

Note also: the current comment uses the same tense, and it was you who's added it :)

Reply:
> I'm using [present perfect](https://en.wikipedia.org/wiki/Present_perfect).

I agree: present perfect is correct here.